cve_bin_tool/checkers/perl.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+"""
+CVE checker for Perl
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-1885/Perl.html
+"""
+from ..util import regex_find
+
+
+def guess_version(lines):
+    regex1 = [
+        r"perl-[45]:([5]+\.[0-9]+\.[0-9]+)",
+        r"perl-[45]:([5]+\.[0-9]+)",
+    ]
+    match = regex_find(lines, *regex1)
+    if match != "UNKNOWN":
+        return match
+    match = re.sub(
+        r"This is perl ([45])+, version ([0-9])+, subversion ([0-9])+",
+        r"\1.\2.\3",
+        lines,
+    )
+    if match:
+        return match
+    else:
+        return "UNKNOWN"
+
+
+def get_version(lines, filename):
+    """
+    returns version information for perl as found in a given file.
+    
+    VPkg: perl, perl
+    """
+    regex_find(lines, *regex)
+    version_info = dict()
+    if filename.startswith("perl"):
+        version_info["is_or_contains"] = "is"
+
+    if "is_or_contains" in version_info:
```

Comment:
You'e never setting "is_or_contains" unless the filename starts with perl, so your test file isn't going to work.  Take a look at #442 https://github.com/intel/cve-bin-tool/pull/442/files to see how we handle this in bluez -- you can probably use the same idea with your guess_version function.